docs/interviewing_options.md

```diff
@@ -1 +1,134 @@
+- If candidate has potentially memorized the solution (from another interview
+  or seen it posted online) this again may not be indicative of their true
+  knowledge.
+- Unfair power imbalance between candidates and interviewers
+
+
+
+## Take-home tests (Homework, Challenges)
+
+Take-home tests are where a candidate is given an assignment to do on their own
+time and typically expected to return their completed assignment back in a set
+period time. this can be hours or days. Sometimes the results of the assignment
+become the topic of conversation during the on-site portion of the interview.
+
+#### Pros
+
+- Allows the candidate to work on their own time, space, environment setup, etc.
+- Candidate can use Google, and other online (legit) resources like they might
+  at a typical job
+- Can be done during non work hours allowing candidates more freedom
+- Less anxiety involved for candidate as no one is watching them
+- Certain types of tests might be conducive to automated testing to check
+  results and validity
+- Again this can be a cheap way to screen candidates as they don't need to come
+  on-site, and act as a stopper for those who are not super serious about
+  actually interviewing
+- Removes the power imbalance that can occur in whiteboard style interview.
+
+#### Cons
+
+- Possibility of candidates plagiarizing work
+- Can bias against people whom have family commitments (caregivers,
+  single parents, etc.) depending on the level of effort needed
+- Again can unintentionally discriminate based on age (favouring those whom
+  might have more free time and can iterate on better solutions)
+- Depending on the complexity of the assignment it may require time for someone
+  to grade and properly evaluate. Depending on the volume of candidates this
+  can be a huge problem.
+
+
+## Bring your own personal code or project
+
+Bringing your own personal code or projects is another method similar to a take
+type of test. The main difference here is that this is something the candidate
+has built vs being told to do. This could be a open source project which they
+have contributed to or maybe a smaller hobby type project. During the interview
+process
+
+#### Pros
+
+- Allows the candidate to work on their own time, space, environment setup, etc.
+- Can be done during non work hours allowing candidates more freedom
+- Less anxiety for the candidate
+- Removes the power imbalance that can occur in whiteboard style interview.
+- Shows the candidates passions outside their core work or job.
+
+#### Cons
+
+- Possibility of candidates plagiarizing work
+- Not everyone has hobby projects which they can show off
+- Again this could indirectly discriminate against some candidates who do not
+  have the time outside of work (family, children, etc.)
+- project may not be relevant to the job at hand therefore making it hard to
+  evaluate the candidates skill set
+
+
+
+## Automated coding challenge
```

Comment:
How is this different from take home? ... it is not clear from this doc.

Reply:
I would say that there are... 

* Coding challenges 
  * Automated versus custom
  * Take home versus on site

Reply:
I think I'm trying to differentiate between a automated test (multiple choice) vs something more open ended where you code something from scratch based on some instructions.